en/sql/function/json_fn.rst

```diff
@@ -477,3 +477,111 @@ The following json path will match all json paths that end with a json array ind
       '2'
       '3'
       NULL
+
+JSON_MERGE_PATCH
+===================================
+
+.. function:: JSON_MERGE_PATCH (json_doc, json_doc [, json_doc] ...)
+The **JSON_MERGE_PATCH** function merges 2 or more json docs and returns the resulting merged json. **JSON_MERGE_PATCH** differs from **JSON_MERGE_PRESERVE** in that it will take the second argument when encountering merging conflicts. **JSON_MERGE_PATCH** is compliant with 
+`RFC 7396 <https://tools.ietf.org/html/rfc7396/>`_.
+
+The merging of 2 json docuemnts is performed after the following rules, recursively:
+
+- when first argument is not an object, the result of the merge is the second object. As an exception, when the second argument is an object, the result of the merge is the merge result of the second argument with an empty object.
+- when 2 objects are merged, the resulting object consists of the following (key, value) pairs:
+
+  - All (key, value) pairs from the first object that have no corresponding (key, value) pairs in the second object.
+  - All (key, value) pairs from the second object that have no corresponding (key, value) pairs in the first object, having values not null.
+  - All (key, value) pairs with same keys that exist in both objects, the second object's pair having the value not null. The values of these pairs become the results of the merging operations performed according to their json type merging rules.
```

Comment:
Please rephrase third case when same key is found in both objects, current explanation is unclear. Something along these lines:
  `If second object value is null ... If second object value is ? ... If both values are json objects, apply rules recursively.`